src/service/resources-impl.js

```diff
@@ -489,13 +422,7 @@ export class ResourcesImpl {
     this.resources_.push(resource);
 
     if (this.intersectionObserver_) {
-      // Wait until upgrade to start observing intersections to give
-      // the browser a chance to layout first. This results in fresher
-      // client rects in the intersection entry, e.g. [overflow] elements
-      // can affect element size since they're `position: relative`.
-      element.whenUpgraded().then(() => {
-        this.intersectionObserver_.observe(resource.element);
-      });
+      this.intersectionObserver_.observe(element);
```

Comment:
Why do we no longer care about waiting for `whenUpgraded`?

Reply:
Similar to removal of other minor edge case handling -- I thought it's better to keep things simple for now to catch the bigger problems first. E.g. this was complicating reasoning about the reparenting race condition. Might need to reintroduce later though.

Reply:
What edge case did it handle? Or was it a perf optimization?